kcp/types.go

```diff
@@ -0,0 +1,24 @@
+package kcp
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// NamespaceReference holds a reference to a specific namespace within a cluster using KCP
+type NamespaceReference struct {
+	// Namespace references a namespace within the cluster
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +required
+	Namespace string `json:"namespace"`
+
+	// Workspace references a KCP workspace within the cluster
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])(":"[a-z0-9]([-a-z0-9]*[a-z0-9]))*$
+	// +optional
+	Workspace string `json:"workspace,omitempty"`
+}
+
+type NamespacedName struct {
```

Comment:
wait what? A NamespacedName is a NamespacedName + a Workspace? That's recursive I am not following. I also only see NamespaceReference used in the code, not NamespacedName

Reply:
It was a goody for you to use in the future if needed. That way you can add more information. I'll remove it and add it back once is needed.